stl/inc/bitset

```diff
@@ -336,29 +337,36 @@ public:
     }
 
     _NODISCARD size_t count() const noexcept { // count number of set bits
-        const char* const _Bitsperbyte = "\0\1\1\2\1\2\2\3\1\2\2\3\2\3\3\4"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\1\2\2\3\2\3\3\4\2\3\3\4\3\4\4\5"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\2\3\3\4\3\4\4\5\3\4\4\5\4\5\5\6"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\3\4\4\5\4\5\5\6\4\5\5\6\5\6\6\7"
-                                         "\4\5\5\6\5\6\6\7\5\6\6\7\6\7\7\x8";
-        const unsigned char* _Ptr       = &reinterpret_cast<const unsigned char&>(_Array);
-        const unsigned char* const _End = _Ptr + sizeof(_Array);
-        size_t _Val                     = 0;
-        for (; _Ptr != _End; ++_Ptr) {
-            _Val += _Bitsperbyte[*_Ptr];
+#if defined(_M_IX86) || defined(_M_X64)
+
+#ifdef __AVX__
+        constexpr bool _Definitely_have_popcnt = true;
+#else
+        const bool _Definitely_have_popcnt = __isa_available >= __ISA_AVAILABLE_SSE42;
+#endif // !defined(__AVX__)
+
+        if (_Definitely_have_popcnt) {
+            size_t _Val = 0;
+            for (size_t _Wpos = 0; _Wpos <= _Words; ++_Wpos) {
+                if constexpr (sizeof(_Ty) == 4) {
+                    _Val += __popcnt(_Array[_Wpos]);
+                } else if constexpr (sizeof(_Ty) == 8) {
+#ifdef _M_IX86
+                    _Val += (__popcnt(_Array[_Wpos] >> 32) + __popcnt(static_cast<unsigned int>(_Array[_Wpos])));
+#else // ^^^ _M_IX86 / !_M_IX86 vvv
+                    _Val += __popcnt64(_Array[_Wpos]);
+#endif // _M_IX86
+                } else {
+                    static_assert(_Always_false<_Ty>, "unexpected underlying type size");
+                }
+            }
+            return _Val;
+        }
+#endif // defined(_M_IX86) || defined(_M_X64)
+        size_t _Val = 0;
+        for (size_t _Wpos = 0; _Wpos <= _Words; ++_Wpos) {
+            _Val += _Popcount_fallback(_Array[_Wpos]);

```

Comment:
It may be worse than before change, especially on 32-bit CPU with 64-bit underlying type

Reply:
If it is, let's fix centralized later